ddl/index.go

```diff
@@ -1809,62 +1808,33 @@ func executeDistGlobalTask(reorgInfo *reorgInfo) error {
 		return err
 	}
 
-	globalTaskManager, err := storage.GetTaskManager()
-	if err != nil {
-		return err
-	}
-
 	taskKey := fmt.Sprintf("ddl/%s/%d", taskType, reorgInfo.Job.ID)
-	globalTask, err := globalTaskManager.GetGlobalTaskByKey(taskKey)
-	if err != nil {
-		return err
-	}
 
-	if globalTask == nil {
-		taskID, err := globalTaskManager.AddNewGlobalTask(taskKey, taskType, distPhysicalTableConcurrency, metaData)
-		if err != nil {
-			return nil
-		}
-
-		globalTask, err = globalTaskManager.GetGlobalTaskByID(taskID)
-		if err != nil {
-			return err
-		}
-
-		if globalTask == nil {
-			return errors.Errorf("cannot find global task with ID %d", taskID)
-		}
-	}
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
 
+	// TODO: handle cancel correctly.
 	ticker := time.NewTicker(CheckBackfillJobFinishInterval)
 	defer ticker.Stop()
-
-	for {
-		<-ticker.C
-		// TODO: handle cancel correctly.
-		if reorgInfo.Job.IsCancelling() {
-			return dbterror.ErrCancelledDDLJob
-		}
-
-		found, err := globalTaskManager.GetGlobalTaskByID(globalTask.ID)
-		if err != nil {
-			logutil.BgLogger().Info("[ddl] get global task error", zap.Int64("taskID", globalTask.ID), zap.Error(err))
-			continue
-		}
-
-		if found == nil {
-			return errors.Errorf("cannot find global task with ID %d", globalTask.ID)
-		}
-
-		if found.State == proto.TaskStateSucceed {
-			return nil
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-ticker.C:
+				if reorgInfo.Job.IsCancelling() {
```

Comment:
I think the `reorgInfo.Job` is never updated during the execution of global task.

Reply:
Just copying from the original logic.